dawn/prototype/atlasGradientDriver.cpp

```diff
@@ -11,8 +11,53 @@
 #include "atlas/output/Gmsh.h"
 #include "atlas_interface.hpp"
 
+#include "AtlasCartesianWrapper.h"
 #include "generated_quadGradient.hpp"
 
+void build_periodic_edges(atlas::Mesh& mesh, int nx, int ny, const AtlasToCartesian& atlasMapper) {
+  atlas::mesh::HybridElements::Connectivity& edgeCellConnectivity =
+      mesh.edges().cell_connectivity();
+  const int missingVal = edgeCellConnectivity.missing_value();
+
+  auto unhash = [](int idx, int nx) -> std::tuple<int, int> {
+    int j = idx / nx;
+    int i = idx - (j * nx);
+    return {i, j};
+  };
+
+  for(int edgeIdx = 0; edgeIdx < mesh.edges().size(); edgeIdx++) {
+    int numNbh = edgeCellConnectivity.cols(edgeIdx);
+    assert(numNbh == 2);
+
+    int nbhLo = edgeCellConnectivity(edgeIdx, 0);
+    int nbhHi = edgeCellConnectivity(edgeIdx, 1);
+
+    assert(!(nbhLo == missingVal && nbhHi == missingVal));
+
+    // if we encountered a missing value, we need to fix the neighbor list
+    if(nbhLo == missingVal || nbhHi == missingVal) {
+      int validIdx = (nbhLo == missingVal) ? nbhHi : nbhLo;
+      auto [cellI, cellJ] = unhash(validIdx, nx);
+      // depending whether we are vertical or horizontal, we need to reflect either the first or
+      // second index
+      if(atlasMapper.edgeOrientation(mesh, edgeIdx) == Orientation::Vertical) {
+        assert(cellI == nx - 1 || cellI == 0);
+        cellI = (cellI == nx - 1) ? 0 : nx - 1;
+      } else { // Orientation::Horizontal
+        assert(cellJ == ny - 1 || cellJ == 0);
+        cellJ = (cellJ == ny - 1) ? 0 : ny - 1;
+      }
+      int oppositeIdx = cellI + cellJ * nx;
+      // ammend the neighbor list
```

Comment:
This won't "merge together" boundary edges which are at the exact opposite side of the grid. I guess just linking to the correct cell neighbors is sufficient for the gradient example, right?